### DISCUSSION TITLE
Deploy storefront NextJs in Azure Devops.

### DISCUSSION THREAD

**Discussion by quang-ng** [medusajs/medusa#4062] [Q&A]
Hello community.

I’m deploying storefront NextJs in Azure DevOps cloud services. I’m facing an issue when I set up env variables  in the Configuration tab in Azure DevOps ( see the image), those variables only access at runtime, but the NextJs need env at build time. So it raises errors when I build the storefront in Azure. Any suggestions on how to setup env variables in Azure for the NextJs project. 

Many thanks.
![Screenshot 2023-05-10 142257](https://github.com/medusajs/medusa/assets/6895574/accdbc9d-01fa-4640-ba33-83cd0d7e8380)

**Comment by quangnst** [comment]
Hi @quang-ng 
Any new updates? We are also facing the same problem and also looking for a solution

**Reply by pevey** [reply]
I'm not familiar with Azure, but on AWS there is something called CodePipeline for the build step and then ECS/CloudFormation for deploy.  Is there not a separate build step on Azure before deploy?

**Comment by pevey** [comment]
https://learn.microsoft.com/en-us/azure/devops/pipelines/create-first-pipeline?view=azure-devops&tabs=java%2Ctfs-2018-2%2Cbrowser

Somewhere in your CI/CD, you need to build the app before you try to deploy it. I've never used Azure, but from googling, the azure way to do this seems to be in the link above.
